### PR TITLE
chore: remove last mentions of Concrete-Numpy

### PR DESCRIPTION
**Pull request by RomanBredehoft**
There is still a method called "cpn_round" in our mixin quantized op but that would create a breaking change, so I added it in https://github.com/zama-ai/concrete-ml-internal/issues/3995 

Also, there are a few (3 I think) links to some concrete-numpy-internal issues left

closes https://github.com/zama-ai/concrete-ml-internal/issues/4074
closes https://github.com/zama-ai/concrete-ml-internal/issues/2250

### CODE REVIEW

File: src/concrete/ml/sklearn/base.py [hunk 1]
```diff
@@ -31,6 +30,9 @@
 from sklearn.linear_model import LinearRegression, LogisticRegression
 from sklearn.utils.validation import check_is_fitted
 
+# pylint: disable-next=ungrouped-imports
+from concrete import fhe as cp
```
Comment (fd0r) [inline src/concrete/ml/sklearn/base.py:34]: why not just use fhe.tag?

**Review by fd0r — APPROVED** [review]

File: src/concrete/ml/sklearn/base.py (hunk 1 above)
Comment (RomanBredehoft) [reply src/concrete/ml/sklearn/base.py:34]: because of some pylint magic 🙂 

that's obviously what I've done at the beginning, which requires us to call `from concrete import fhe`. But guess what, since we have this `fhe` parameter in our `predict` methods, pylint was throwing a `W0621: Redefining name 'fhe' from outer scope (line 33) (redefined-outer-name)` 🙂 (https://github.com/zama-ai/concrete-ml/actions/runs/7463742016/job/20309199710)

and I didn't want to have `from concrete.fhe import tag, zeros, array` because calling `zeros(labels.shape)` or `array(arr)` in or code feels a bit ambiguous !

Comment (fd0r) [reply src/concrete/ml/sklearn/base.py:34]: Makes sense thanks a lot!
